src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String SUPPLY_NAME = "supply";
+    static final String BUY_NAME = "buy";
+    static final String RESULT_NAME = "result";
+    static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String operation;
+        int amount;
+        String[] linePart;
+        StringBuilder report = new StringBuilder();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+
+            while (value != null) {
+                linePart = value.split(COMMA);
+                if (linePart.length == 2) {
+                    operation = linePart[0].trim();
+                    amount = Integer.parseInt(linePart[1].trim());
+                    if (operation.equals(SUPPLY_NAME)) {
+                        supplyAmount += amount;
+                    } else if (operation.equals(BUY_NAME)) {
+                        buyAmount += amount;
+                    } else {
+                        throw new RuntimeException("Operation does not exist");
+                    }
+                } else {
+                    throw new RuntimeException("Line incorrect");
```

Malformed-line handling currently throws a generic `RuntimeException("Line incorrect")`. The checklist requires informative exception messages and not ignoring exceptions. Quote: "Don't ignore exceptions ... Better re-throw `RuntimeException` with the original exception and some message in the parameters:" Instead of a generic message, include context such as `fromFileName`, the line content, and the line number. See checklist .